### DISCUSSION TITLE
Error (use-package): marginalia/:init: Symbol’s function definition is void: marginalia-mode

### DISCUSSION THREAD

**Discussion by pointerto** [minad/marginalia#167] [Q&A] [answered]
I followed the config from manual, but an ERROR arised as below:

> Error (use-package): marginalia/:init: Symbol’s function definition is void: marginalia-mode

**Comment by minad** [comment, marked as answer]
This is likely an installation problem. Try `M-x list-packages RET` and then (re-)install Marginalia.

**Reply by pointerto** [reply]
I manage packages by use-package manually. The issue has been solved with modified config as below and the package works well till now.
`(use-package marginalia
  :config
  (marginalia-mode))`

**Reply by minad** [reply]
No, using :config is not correct here, since then Marginalia will not be loaded and `marginalia-mode` will not be enabled right away. The README explicitly states:

~~~ elisp
  ;; Marginalia must be activated in the :init section of use-package such that
  ;; the mode gets enabled right away. Note that this forces loading the
  ;; package.
~~~

It could be that you are not managing your packages correctly and miss to generate or load the autoloads.

**Reply by pointerto** [reply]
I tried `update-directory-autoloads` in the package directory, which to generate a xxx-autoloads.el there (This method is useful for another package I installed before). Then I used README manaual config again, but it doesn't work and arised an error same to showed in disscussion title

After failing above I added `:command marginalia-mode` into README config, the package works normally this time. Is this a correct way to install marginalia?
```
;; Enable rich annotations using the Marginalia package
(use-package marginalia
  **:commands marginalia-mode**
  ;; Bind `marginalia-cycle' locally in the minibuffer.  To make the binding
  ;; available in the *Completions* buffer, add it to the
  ;; `completion-list-mode-map'.
  :bind (:map minibuffer-local-map
         ("M-A" . marginalia-cycle))

  ;; The :init section is always executed.
  :init

  ;; Marginalia must be activated in the :init section of use-package such that
  ;; the mode gets enabled right away. Note that this forces loading the
  ;; package.
  (marginalia-mode))
```

**Reply by minad** [reply]
> Is this a correct way to install marginalia?

No, the correct way is to install Marginalia via the builtin Emacs package manager (`M-x package-install`), such that the Elisp files are properly compiled and autoloads are generated correctly. It seems to be that your installation method does not make sure that autoloads are loaded (or even generated). As alternatives to the builtin package manager, there exist package managers like Straight or Elpaca, with their own unique advantages, but they also introduce their own difficulties.

**Comment by jcfk** [comment]
The problem was solved for me by adding `:ensure t` first:

```
(use-package marginalia
  :ensure t
  :bind (:map minibuffer-local-map
         ("M-A" . marginalia-cycle))
  :init
  (marginalia-mode))
```

**Reply by harveyz0** [reply]
me too
